src/parser/TurtleParser.h

```diff
@@ -440,3 +281,104 @@ class TurtleParser {
   FRIEND_TEST(TurtleParserTest, blankNode);
   FRIEND_TEST(TurtleParserTest, blankNodePropertyList);
 };
+
+/**
+ * This class is a TurtleParser that always assumes that
+ * its input file is an uncompressed .ttl file that will be read in
+ * chunks. Input file can also be a stream like stdin.
+ */
+class TurtleStreamParser : public TurtleParser {
+  // struct that can store the state of a parser
+  // the previously extracted triples are not stored
+  // but only the number of triples that were already present
+  // before the backup
+  struct TurtleParserBackupState {
+    ad_utility::HashMap<std::string, std::string> _blankNodeMap;
+    size_t _numBlankNodes = 0;
+    size_t _numTriples;
+    const char* _tokenizerPosition;
+    size_t _tokenizerSize;
+  };
+
+ public:
+  // Default construction needed for tests
+  TurtleStreamParser() = default;
+  TurtleStreamParser(const string& filename) {
+    LOG(INFO) << "Initialize turtle Parsing from uncompressed file or stream "
+              << filename << '\n';
+    initialize(filename);
+  }
+
+  // inherit the wrapper overload
+  using TurtleParser::getLine;
+
+  virtual bool getLine(std::array<string, 3>* triple) override;
+
+  virtual void initialize(const string& filename) override;
+
+ private:
+  // Backup the current state of the turtle parser to a
+  // TurtleparserBackupState object
+  // This can be used e.g. when parsing from a compressed input
+  // and the currently uncompressed buffer is not sufficient to parse the
+  // next expression
+  TurtleParserBackupState backupState() const;
+
+  // Reset the parser to the state indicated by the argument
+  // Must be called on the same parser object that was used to create the backup
+  // state (the actual triples are not backed up)
+  bool resetStateAndRead(const TurtleParserBackupState state);
+
+  // used with compressed inputs (a certain number of bytes is uncompressed at
+  // once, might also be cut in the middle of a utf8 string. (bzip2 does not no
```

Comment:
`s/no/know/g`

Reply:
the whole comment did not make too much sense without bzip2, I reformulated.